core/transfer/src/main/java/org/eclipse/dataspaceconnector/transfer/core/transfer/AsyncTransferProcessManager.java

```diff
@@ -123,43 +125,8 @@ public TransferInitiateResult initiateProviderRequest(DataRequest dataRequest) {
     }
 
     @Override
-    public void transitionRequestAck(String processId) {
-        TransferProcess transferProcess = transferProcessStore.find(processId);
-        transferProcess.transitionRequestAck();
-        transferProcessStore.update(transferProcess);
-    }
-
-    @Override
-    public void transitionProvisioned(String processId) {
-        TransferProcess transferProcess = transferProcessStore.find(processId);
-        transferProcess.transitionProvisioned();
-        transferProcessStore.update(transferProcess);
-    }
-
-    @Override
-    public void transitionError(String processId, String detail) {
-        TransferProcess transferProcess = transferProcessStore.find(processId);
-        transferProcess.transitionError(detail);
-        transferProcessStore.update(transferProcess);
-    }
-
-    @Override
-    public Result<TransferProcessStates> deprovision(String processId) {
-        var process = transferProcessStore.find(processId);
-        if (process == null) {
-            return Result.failure("not found");
-        }
-
-        if (Set.of(DEPROVISIONED.code(), DEPROVISIONING_REQ.code(), DEPROVISIONING.code(), ENDED.code()).contains(process.getState())) {
-            monitor.info("Request already deprovisioning or deprovisioned.");
-        } else {
-            monitor.info("starting to deprovision data request " + processId);
-            process.transitionCompleted();
-            process.transitionDeprovisionRequested();
-            transferProcessStore.update(process);
-        }
-
-        return Result.success(TransferProcessStates.from(process.getState()));
+    public void enqueueCommand(Command command) {
```

Comment:
Have you considered implementing lifecycle operations in the `TransferProcessManager` (provisionInitialProcesses, sendOrProcessProvisionedRequests, ...) or at least parts of it as commands too?

Reply:
yes we have, and we decided against it for various reasons. chief amongst those are that the "normal" state machine transitions should not be influenced by the command queue. We also want to guarantee fairness and that the SM advances, even it there are loads of commands in the queue.